docs/transactions.md

```diff
@@ -0,0 +1,166 @@
+`.submit()` method, or as a group by calling `submit_inner_txn()`
+
+
+### Examples
+
+#### Create and submit an inner transaction
+```python
+import puyapy
+
+
+@puyapy.subroutine
+def create_payment(amount: puyapy.UInt64, receiver: puyapy.Account) -> None:
+    puyapy.itxn.PaymentTransactionParams(
+        amount=amount,
+        receiver=receiver,
+        fee=0,
+    ).submit()
+```
+
+#### Accessing result of a submitted inner transaction
+```python
+import puyapy
+
+
+@puyapy.subroutine
+def create_asset() -> puyapy.Asset:
+    asset_txn = puyapy.itxn.AssetConfigTransactionParams(
+        asset_name=b"Puya",
+        unit_name=b"PYA",
+        total=1000,
+        decimals=3,
+        fee=0,
+    ).submit()
+    return asset_txn.created_asset
+```
+
+#### Submitting multiple transactions
+```python
+import puyapy
+
+
+@puyapy.subroutine
+def create_two_assets() -> tuple[puyapy.Asset, puyapy.Asset]:
+    asset1_params = puyapy.itxn.AssetConfigTransactionParams(
+        asset_name=b"Puya",
+        unit_name=b"PYA",
+        total=1000,
+        decimals=3,
+        fee=0,
+    )
+    asset2_params = asset1_params.copy() # a parameter object can be copied and then modified
+    asset2_params.set(asset_name=b"Puya 2", unit_name="PYA2")
+    asset1_txn, asset2_txn = puyapy.itxn.submit_inner_txn(asset1_params, asset2_params)
+    return asset1_txn.created_asset, asset2_txn.created_asset
+```
+
+#### Create an application, and then call it
+```python
+import puyapy
+
+def get_program_bytes() -> tuple[puyapy.Bytes, puyapy.Bytes]:
+    ...
+
+@puyapy.subroutine
+def create_application() -> puyapy.Application:
+    approval, clear = get_program_bytes()
+    application_txn = puyapy.itxn.ApplicationCallTransactionParams(
+        approval_program=approval,
+        clear_state_program=clear,
+    ).submit()
+    puyapy.itxn.ApplicationCallTransactionParams(
+        application_id=application_txn.created_application,
+        application_args=(puyapy.Bytes(b"arg1"), puyapy.Bytes(b"arg2"))
+    ).submit()
+    return application_txn.created_application
+```
+
+#### Create and submit transactions in a loop
+```python
+import puyapy
+
+@puyapy.subroutine
+def create_payments(receivers: tuple[puyapy.Account, puyapy.Account, puyapy.Account]) -> None:
+    for receiver in receivers:
+        puyapy.itxn.PaymentTransactionParams(
+            amount=puyapy.UInt64(1_000_000),
+            receiver=receiver,
+            fee=0,
+        ).submit()
+```
+### Limitations
+
+Inner transactions are powerful, but currently do have some restrictions in how they are used.
+
+* Inner transaction objects cannot be passed to subroutines
+  ```{note}
+  Individual fields can be passed to subroutines
+  ```
+* Inner transaction parameters cannot be reassigned without a `.copy()`
+* Inner transactions cannot be reassigned
+* Inner transactions array values cannot be accessed if there is a subsequent inner transaction submitted. 
```

Comment:
I'm not quite sure what this means - maybe an example is in order

Reply:
Added examples for all of these scenarios